sdl_android/src/main/java/com/smartdevicelink/transport/SdlRouterService.java

```diff
@@ -2044,6 +2034,28 @@ private void stopClientPings(){
 		}
 		pingIntent = null;
 	}
+
+	/**
+	 * Iterate through all of the apps that we know are listening for this intent
+	 * with an explicit intent (necessary for Android O SDK 26+)
+	 *
+	 * @param intent - the intent to send explicitly
+	 * @param isRepeatingPings - If this is coming from a method that will repeat pings,
+	 * we will use a cached list instead of querying each ping
+	 */
+	private void sendExplicitBroadcast(Intent intent, Boolean isRepeatingPings) {
```

Comment:
Instead of a boolean here, why don't we pass a list of ResolveInfo that we want to send the intents to. If the list is null, we do the normal query of broadcast receivers.  Then for the ping runnable you can cache the list in the runnable itself.